on 3 machines it broke, on one it succeeded, here are the logs

```
numpyro> FAILED test/contrib/test_enum_elbo.py::test_guide_plate_contraction
numpyro> FAILED test/infer/test_autoguide.py::test_beta_bernoulli[AutoGuideList]
numpyro> FAILED test/infer/test_hmc_util.py::test_gaussian_subposterior[True-consensus]
numpyro> FAILED test/infer/test_mcmc.py::test_SA_chain_gradient_free[2]
numpyro> FAILED test/infer/test_reparam.py::test_loc_scale[1-Normal-1.0-(3, 2)]
numpyro> FAILED test/infer/test_mcmc.py::test_logistic_regression_x64[SA]
numpyro> FAILED test/infer/test_svi.py::test_renyi_nonnested_plates
numpyro> FAILED test/infer/test_svi.py::test_tracegraph_gaussian_chain[5-6000-0.003-0.05-0.6]
numpyro> FAILED test/infer/test_mcmc.py::test_chain[parallel-True]
numpyro> FAILED test/infer/test_autoguide.py::test_autosldais
numpyro> FAILED test/infer/test_autoguide.py::test_logistic_regression[Trace_ELBO-AutoLaplaceApproximation]
numpyro> FAILED test/infer/test_mcmc.py::test_chain[vectorized-False]
numpyro> FAILED test/infer/test_mcmc.py::test_beta_bernoulli_x64[NUTS]
numpyro> FAILED test/infer/test_mcmc.py::test_logistic_regression_x64[AIES]
numpyro> FAILED test/infer/test_hmc_util.py::test_welford_covariance[False-True-True]
```

```
numpyro> FAILED test/infer/test_mcmc.py::test_logistic_regression_x64[BarkerMH]
numpyro> FAILED test/infer/test_svi.py::test_beta_bernoulli[optimizer0-elbo1]
numpyro> FAILED test/infer/test_infer_util.py::test_predictive[True]
numpyro> FAILED test/test_nn.py::test_masks[1-1-6]
numpyro> FAILED test/infer/test_mcmc.py::test_logistic_regression_x64[AIES]
```

```
numpyro> FAILED test/contrib/test_enum_elbo.py::test_guide_plate_contraction
numpyro> FAILED test/infer/test_infer_util.py::test_initialize_model_change_point[init_to_median]
numpyro> FAILED test/infer/test_autoguide.py::test_beta_bernoulli[AutoGuideList]
numpyro> FAILED test/infer/test_hmc_gibbs.py::test_hmcecs_normal_normal[2-50-2-NUTS]
numpyro> FAILED test/infer/test_mcmc.py::test_chain[parallel-False]
numpyro> FAILED test/infer/test_infer_util.py::test_initialize_model_change_point[init_strategy0]
numpyro> FAILED test/infer/test_svi.py::test_vectorized_particle
numpyro> FAILED test/test_handlers.py::test_plate[model_subsample_1]
numpyro> FAILED test/infer/test_autoguide.py::test_autosldais
numpyro> FAILED test/infer/test_svi.py::test_mutable_state[Trace_ELBO-10-False]
numpyro> FAILED test/infer/test_mcmc.py::test_logistic_regression_x64[SA]
numpyro> FAILED test/infer/test_mcmc.py::test_logistic_regression_x64[AIES]
numpyro> FAILED test/infer/test_mcmc.py::test_logistic_regression_x64[ESS]
```

--- a/pkgs/development/python-modules/numpyro/default.nix
+++ b/pkgs/development/python-modules/numpyro/default.nix
@@ -1,7 +1,7 @@
 {
   lib,
   buildPythonPackage,
-  fetchPypi,
+  fetchFromGitHub,
 
   # build-system
   setuptools,
@@ -13,11 +13,19 @@
   numpy,
   tqdm,
 
-  # checks
+  # tests
+  # Our current version of tensorflow (2.13.0) is too old and doesn't support python>=3.12
+  # We remove optional test dependencies that require tensorflow and skip the corresponding tests to
+  # avoid introducing a useless incompatibility with python 3.12:
+  # dm-haiku,
+  # flax,
+  # tensorflow-probability,
   funsor,
+  graphviz,
+  optax,
+  pyro-api,
   pytestCheckHook,
-# TODO: uncomment when tensorflow-probability gets fixed.
-# , tensorflow-probability
+  scikit-learn,
 }:
 
 buildPythonPackage rec {
@@ -25,9 +33,11 @@ buildPythonPackage rec {
   version = "0.15.3";
   pyproject = true;
 
-  src = fetchPypi {
-    inherit version pname;
-    hash = "sha256-9EXurmIA+IPXkNZc4p/yRfepY5usMyLZk+zPkcRAI7M=";
+  src = fetchFromGitHub {
+    owner = "pyro-ppl";
+    repo = "numpyro";
+    rev = "refs/tags/${version}";
+    hash = "sha256-g+ep221hhLbCjQasKpiEAXkygI5A3Hglqo1tV8lv5eg=";
   };
 
   build-system = [ setuptools ];
@@ -41,9 +51,14 @@ buildPythonPackage rec {
   ];
 
   nativeCheckInputs = [
+    # dm-haiku
+    # flax
     funsor
+    graphviz
+    optax
+    pyro-api
     pytestCheckHook
-    # TODO: uncomment when tensorflow-probability gets fixed.
+    scikit-learn
     # tensorflow-probability
   ];
 
@@ -60,17 +75,45 @@ buildPythonPackage rec {
     "test_kl_dirichlet_dirichlet"
     "test_kl_univariate"
     "test_mean_var"
+
     # Tests want to download data
     "data_load"
     "test_jsb_chorales"
+
     # RuntimeWarning: overflow encountered in cast
     "test_zero_inflated_logits_probs_agree"
+
     # NameError: unbound axis name: _provenance
     "test_model_transformation"
+
+    # require dm-haiku
+    "test_flax_state_dropout_smoke"
+    "test_flax_module"
+    "test_random_module_mcmc"
+
+    # require flax
+    "test_haiku_state_dropout_smoke"
+    "test_haiku_module"
+    "test_random_module_mcmc"
+
+    # require tensorflow-probability
+    "test_modified_bessel_first_kind_vect"
+    "test_diag_spectral_density_periodic"
+    "test_kernel_approx_periodic"
+    "test_modified_bessel_first_kind_one_dim"
+    "test_modified_bessel_first_kind_vect"
+    "test_periodic_gp_one_dim_model"
+    "test_no_tracer_leak_at_lazy_property_sample"
   ];
 
-  # TODO: remove when tensorflow-probability gets fixed.
-  disabledTestPaths = [ "test/test_distributions.py" ];
+  disabledTestPaths = [
+    # require jaxns (unpackaged)
+    "test/contrib/test_nested_sampling.py"
+
+    # requires tensorflow-probability
+    "test/contrib/test_tfp.py"
+    "test/test_distributions.py"
+  ];
 
   meta = {
     description = "Library for probabilistic programming with NumPy";

--- a/pkgs/development/python-modules/numpyro/default.nix
+++ b/pkgs/development/python-modules/numpyro/default.nix
@@ -104,6 +104,11 @@ buildPythonPackage rec {
     "test_modified_bessel_first_kind_vect"
     "test_periodic_gp_one_dim_model"
     "test_no_tracer_leak_at_lazy_property_sample"
+
+    # flaky on darwin
+    # TODO: uncomment at next release (0.15.4) as it has been fixed:
+    # https://github.com/pyro-ppl/numpyro/pull/1863
+    "test_change_point_x64"
   ];
 
   disabledTestPaths = [

